src/test/ledger/Invariants_test.cpp

```diff
@@ -215,6 +229,176 @@ class Invariants_test : public beast::unit_test::suite
             });
     }
 
+    void
+    testXRPBalanceCheck(bool enabled)
+    {
+        using namespace test::jtx;
+        testcase << "checks " << (enabled ? "enabled" : "disabled") <<
+            " - XRP balance checks";
+
+        doInvariantCheck (enabled,
+            {{ "Cannot return non-native STAmount as XRPAmount" }},
```

Comment:
I just noticed it when figuring out where this error message is coming from.
https://github.com/ripple/rippled/blob/develop/src/ripple/protocol/impl/STAmount.cpp#L302
It seemed like this test was more checking that `Throw` than the the invariant check itself.
I wasn't sure if we wanted the same for the `iou()` case.